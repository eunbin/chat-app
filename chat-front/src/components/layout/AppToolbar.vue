<template>
  <v-toolbar
    color="primary"
    dark
    flat
    scroll-off-screen
    scroll-target="#scrolling-techniques"
  >
    <slot></slot>

    <v-toolbar-title>{{ title }}</v-toolbar-title>

    <v-spacer></v-spacer>

    <slot name="right"></slot>
  </v-toolbar>
</template>

<script>
export default {
  name: 'AppToolbar',
  props: {
    title: {
      type: String,
      default: '채팅'
    }
  }
}
</script>
