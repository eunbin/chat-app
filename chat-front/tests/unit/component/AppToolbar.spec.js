import { mount } from '@vue/test-utils'
import '../test-setup'
import AppToolbar from '@/components/layout/AppToolbar.vue'

describe('AppToolbar.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(AppToolbar)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it('snapshot check with roomInfo', () => {
    const roomInfo = {
      roomId: 1,
      title: 'Hello Chat Room',
      members: []
    }
    const wrapper = mount(AppToolbar, {
      propsData: { title: roomInfo.title }
    })

    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it('given roomInfo when enter chat room then show title in toolbar', () => {
    const roomInfo = {
      roomId: 1,
      title: 'Hello Chat Room',
      members: []
    }
    const wrapper = mount(AppToolbar, {
      propsData: { title: roomInfo.title }
    })

    wrapper.find('.v-toolbar')
    expect(wrapper.text()).toEqual(roomInfo.title)
  })
})
