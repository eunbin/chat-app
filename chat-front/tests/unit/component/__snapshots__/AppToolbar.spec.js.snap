// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToolbar.vue snapshot check with items 1`] = `
<nav
  class="v-toolbar elevation-0 theme--dark primary"
  style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
>
  <div
    class="v-toolbar__content"
    style="height: 64px;"
  >
     
    <div
      class="v-toolbar__title"
    >
      Hello Chat Room
    </div>
     
    <div
      class="spacer"
    />
     
  </div>
</nav>
`;
